src/main/scala/fs2/Process1.scala

```diff
@@ -77,6 +77,24 @@ object process1 {
     _ repeatPull { _.await.flatMap { case chunk #: h => Pull.output(f(chunk)) as h }}
 
   /**
+    * Maps a running total according to `S` and the input with the function `f`.
+    *
+    * @example {{{
+    * scala> Stream("Hello", "World")
+    *      |   .mapAccumulate(0)((l, s) => (l + s.length, s.head))
+    *      |   .covary[util.Task].runLog.run.run
+    * res0: Vector[(Int, Char)] = Vector((5,H), (10,W))
+    * }}}
+    */
+  def mapAccumulate[F[_],S,I,O](init: S)(f: (S,I) => (S,O)): Stream[F,I] => Stream[F,(S,O)] =
+    _ pull { _.await1.flatMap { case chunk #: h =>
```

Comment:
Actually, I just noticed you are using `await1`. Let's preserve chunkiness. Add a `mapAccumulate` to `Chunk`:

`def mapAccumulate[S](s0: S)(f: (S,A) => (S,B): (S,Chunk[B])`

Maybe a `filterMapAccumulate` that returns an `Option[B]` on each step also, combining filtering.

Then define `process1.mapAccumulate` in terms of this.